src/middlewares/upload.middleware.ts

```diff
@@ -35,20 +44,56 @@ export const uploadMiddleware = multer({
       const date = getFormattedDate();
       const uuid = uuidv4();
       const originalName = file.originalname;
-      const ext = path.extname(originalName); // 확장자
-      const baseName = path.basename(originalName, ext); // 확장자를 제외한 파일 이름
+      const ext = path.extname(originalName);
+      const baseName = path.basename(originalName, ext);
       cb(null, `uploads/${date}/${uuid}_${baseName}${ext}`);
     },
   }),
   fileFilter: (req: Request, file, cb) => {
-    // 이미지 파일만 허용
-    if (file.mimetype.startsWith('image/')) {
-      cb(null, true);
-    } else {
-      cb(HttpException.badRequest('이미지 파일만 업로드 가능합니다.'));
+    const ext = path.extname(file.originalname).toLowerCase().substring(1);
+
+    // 1. 허용된 확장자인지 검증
+    if (!ALLOWED_EXTENSIONS.includes(ext)) {
+      return cb(HttpException.badRequest(`허용되지 않는 확장자입니다: ${ext}`));
+    }
+
+    // 2. 허용된 MIME 타입인지 검증
+    const allowedMimeTypes = Object.keys(MIME_TYPE_MAP);
+    if (!allowedMimeTypes.includes(file.mimetype)) {
+      return cb(HttpException.badRequest('허용되지 않는 파일 유형입니다.'));
+    }
+
+    // 3. 확장자와 MIME 타입이 일치하는지 검증
+    if (!MIME_TYPE_MAP[file.mimetype]?.includes(ext)) {
+      return cb(HttpException.badRequest('파일 확장자와 실제 파일 유형이 일치하지 않습니다.'));
     }
+
+    cb(null, true);
   },
   limits: {
     fileSize: 5 * 1024 * 1024, // 5MB 제한
   },
 });
+
+// Multer 에러를 핸들링하는 커스텀 미들웨어
+export const uploadImage = (req: Request, res: Response, next: NextFunction) => {
+  upload.single('image')(req, res, (err) => {
+    if (err instanceof MulterError) {
+      switch (err.code) {
+        case 'LIMIT_FILE_SIZE':
+          return next(HttpException.badRequest('파일 크기는 5MB를 초과할 수 없습니다.'));
+        case 'LIMIT_UNEXPECTED_FILE':
+          return next(HttpException.badRequest('하나의 파일만 업로드할 수 있습니다.'));
+        default:
+          // 기타 Multer 에러
+          return next(HttpException.badRequest(`파일 업로드 중 오류가 발생했습니다: ${err.message}`));
```

Comment:
```suggestion
          return next(HttpException.badRequest(`파일 업로드 중 오류가 발생했습니다`));
```
사용자 응답에 에러 메시지를 그대로 내보내면 보안상 안 좋을 것 같기도해서, 자세한 에러는 로그로 저장하고 사용자한테는 일반적인 응답을 내보내는게 어떨까 합니다